### PR TITLE
ci(jenkins): remove test output, remove test retries, minor refactor [WIP]

### PR DESCRIPTION
**Pull request by kuisathaverat**
We want to test how much the speed improves by removing the stdout

* remove test output
* minor refactor

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -165,12 +165,20 @@ def generateStep(version, tav = ''){
...
         }
         junit(allowEmptyResults: true, keepLongStdio: true, testResults: "${BASE_DIR}/**/junit-*.xml")
         codecov(repo: 'apm-agent-nodejs', basedir: "${BASE_DIR}", secret: "${CODECOV_SECRET}")
       }
     }
   }
 }
+
+def runNodejs(Closure body){
+  docker.image('node:11').inside(){
```
Comment (watson) [inline Jenkinsfile]: Does the `11` part of `node:11` mean Node.js version 11?

If so we might want to use either v10 or v12 as those are or will be LTS and is considered more stable (I know this might not be part of this PR, but just wanted to mention it as I saw it here 😄)

Comment (kuisathaverat) [reply Jenkinsfile]: ok, 12 in that case

**Comment by kuisathaverat** [comment]
after a few build, the time is consistent on 1h 23 min, checking with the master branch it is about 20 min less

**Comment by Qard** [comment]
What's the status of this. Appears to be failing tests. Does it just need a rebase?

**Comment by kuisathaverat** [comment]
@Qard This PR is WIP, it is a performance test, we merge with master on every build, those test are failing on master too since 6 days ago https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/master/46/pipeline , it is failing to make the `npm install` on Node.js 6 with a weird error related with zlib

```
[2019-07-09T05:56:35.984Z] node_tests_1     | npm timing stage:rollbackFailedOptional Completed in 5ms
[2019-07-09T05:56:35.984Z] node_tests_1     | npm timing stage:runTopLevelLifecycles Completed in 15326ms
[2019-07-09T05:56:36.246Z] node_tests_1     | npm ERR! zlib: zlib binding closed
[2019-07-09T05:56:36.246Z] node_tests_1     | npm timing npm Completed in 16137ms
[2019-07-09T05:56:36.246Z] node_tests_1     | 
[2019-07-09T05:56:36.246Z] node_tests_1     | npm ERR! A complete log of this run can be found in:
[2019-07-09T05:56:36.246Z] node_tests_1     | npm ERR!     /tmp/.npm/_logs/2019-07-09T05_56_36_038Z-debug.log
```

I'll test it locally to see if I saw the same error, Node.js 6 is a PITA

**Comment by kuisathaverat** [comment]
I've reviewed the build times and there is no significant change to build times after removing the stdout so it does not worth

**Comment by kuisathaverat** [comment]
@Qard Node.js 6 is failing to install `http-signature`, locally with docker faile in the same package, outside Docker fails on an optional dependency and can continue, it is a little bit erratic. So I have forced to install npm@6.9.0 (latest is 6.10.0) and it works so there is something that breaks the `npm install` on Node.js and npm@6.10.0 

```
node_tests_1     | 801 http fetch GET 304 https://registry.npmjs.org/http-signature/-/http-signature-1.2.0.tgz 84ms (from cache)
node_tests_1     | 802 silly fetchPackageMetaData error for http-signature@^1.2.0 zlib: zlib binding closed```
